x-pack/plugins/apm/public/components/app/correlations/failed_transactions_correlations.tsx

```diff
@@ -56,78 +61,33 @@ export function FailedTransactionsCorrelations({
   } = useApmPluginContext();
   const trackApmEvent = useUiTracker({ app: 'apm' });
 
-  const { serviceName, transactionType } = useApmServiceContext();
-
-  const {
-    query: { kuery, environment, rangeFrom, rangeTo },
-  } = useApmParams('/services/:serviceName');
-
-  const { urlParams } = useUrlParams();
-  const { transactionName } = urlParams;
-
-  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
-
   const inspectEnabled = uiSettings.get<boolean>(enableInspectEsQueries);
 
-  const result = useFailedTransactionsCorrelationsFetcher();
-
-  const {
-    ccsWarning,
-    log,
-    error,
-    isRunning,
-    progress,
-    startFetch,
-    cancelFetch,
-  } = result;
-
-  const startFetchHandler = useCallback(() => {
-    startFetch({
-      environment,
-      kuery,
-      serviceName,
-      transactionName,
-      transactionType,
-      start,
-      end,
-    });
-  }, [
-    startFetch,
-    environment,
-    serviceName,
-    transactionName,
-    transactionType,
-    kuery,
-    start,
-    end,
-  ]);
-
-  useEffect(() => {
-    startFetchHandler();
-    return cancelFetch;
-  }, [cancelFetch, startFetchHandler]);
+  const { progress, response, startFetch, cancelFetch } = useSearchStrategy(
+    APM_SEARCH_STRATEGIES.APM_FAILED_TRANSACTIONS_CORRELATIONS
+  );
+  const progressNormalized = progress.loaded / progress.total;
 
   const [
     selectedSignificantTerm,
     setSelectedSignificantTerm,
-  ] = useState<FailedTransactionsCorrelationValue | null>(null);
+  ] = useState<FailedTransactionsCorrelation | null>(null);
 
   const selectedTerm = useMemo(() => {
     if (selectedSignificantTerm) return selectedSignificantTerm;
-    return result?.values &&
-      Array.isArray(result.values) &&
-      result.values.length > 0
-      ? result?.values[0]
+    return Array.isArray(response.failedTransactionsCorrelations) &&
+      response.failedTransactionsCorrelations.length > 0
+      ? response.failedTransactionsCorrelations[0]
       : undefined;
-  }, [selectedSignificantTerm, result]);
+  }, [selectedSignificantTerm, response.failedTransactionsCorrelations]);
```

Comment:
Afaict this doesn't need to be memoized:


```ts
const selectedTerm = selectedSignificantTerm ?? response.failedTransactionsCorrelations?.[0]
```

Reply:
Updated in b66541336af.